pkg/shared/util/volume.go

```diff
@@ -30,9 +30,20 @@ var (
 	configMapKeySelectorType = reflect.TypeOf(&corev1.ConfigMapKeySelector{})
 )
 
+type GetSecrets interface {
+	GetSecretFromVolume(selector *corev1.SecretKeySelector) (string, error)
+	GetConfigMapFromVolume(selector *corev1.ConfigMapKeySelector) (string, error)
+}
+
+type OsFile struct{}
+
+func GetSecretFromVolume(selector *corev1.SecretKeySelector) (string, error) {
+	return OsFile{}.GetSecretFromVolume(selector)
+}
+
 // GetSecretFromVolume retrieves the value of mounted secret volume
 // "/var/numaflow/secrets/${secretRef.name}/${secretRef.key}" is expected to be the file path
-func GetSecretFromVolume(selector *corev1.SecretKeySelector) (string, error) {
+func (_ OsFile) GetSecretFromVolume(selector *corev1.SecretKeySelector) (string, error) {
```

Comment:
Using underscore will bring warning in some IDEs.

Reply:
>Using underscore will bring warning in some IDEs.
I thought the underscore was required, changed!